From 86d0ad265a1a0ea91824cbfa6f8ee70a7d797786 Mon Sep 17 00:00:00 2001
From: Jussi Laako <jussi@sonarnerd.net>
Date: Mon, 26 Oct 2020 20:56:58 +0200
Subject: [PATCH] ALSA: usb-audio: Add Cayin, FiiO and Khadas VID to DSD
 detection whitelist

Cayin, FiiO and Khadas DACs have DSD auto-detection support.

Signed-off-by: Jussi Laako <jussi@sonarnerd.net>
---
 sound/usb/quirks.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/usb/quirks.c b/sound/usb/quirks.c
index 9f63164d09bd..e252b559156b 100644
--- a/sound/usb/quirks.c
+++ b/sound/usb/quirks.c
@@ -1738,7 +1738,10 @@ u64 snd_usb_interface_dsd_format_quirks(struct snd_usb_audio *chip,
 	case 0x25ce:  /* Mytek devices */
 	case 0x278b:  /* Rotel? */
 	case 0x292b:  /* Gustard/Ess based devices */
+	case 0x2972:  /* FiiO devices */
 	case 0x2ab6:  /* T+A devices */
+	case 0x2d87:  /* Cayin? */
+	case 0x3353:  /* Khadas devices */
 	case 0x3842:  /* EVGA */
 	case 0xc502:  /* HiBy devices */
 		if (fp->dsd_raw)
-- 
2.34.1

